### PR TITLE
chore(#320): room type

### LINKED ISSUE #320: add a room type to each room
add a room type in the UI and in the database

### What needs to be done
add a new collum in the database for the room type 
add a new variable attached to each room called room type
make the admin able to set the room type when creating a room

### PR DESCRIPTION
**Pull request by joeyabou**
Added a room type variable to rooms. Modified the create room UI to reflect this change and added the collum of room type to the database.

### Test plan (required)

image

### Closing issues
- closes #320 

### Sub-tasks Checklist
- [X] UI Implementation
- [ ] UI Test
- [X] Backend Implementation
- [ ] Backend Tests
- [ ] Integration Tests
- [ ] System Tests
- [ ] Documentation

### CODE REVIEW

File: database/migrations/2021_01_20_031335_add_room_type_to_rooms_table.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('rooms', function (Blueprint $table) {
+            $table->string('room_type')->default('');
+        });
```
Comment (EvanDime) [inline database/migrations/2021_01_20_031335_add_room_type_to_rooms_table.php:18]: if you have a const list of room types, why is the default empty string?

File: resources/js/Pages/Admin/Rooms/CreateRoomForm.vue [hunk 2]
```diff
@@ -76,6 +76,28 @@
...
+                                name="room_type"
+                                id="room_type"
+                              >
+                                <option value="" selected disabled hidden>
+                                  Select Room Type
+                                </option>
+                                <option value="lounge">Lounge</option>
+                                <option value="mezzanine">Mezzanine</option>
+                                <option value="conference">Conference</option>
+                              </select>
```
Comment (EvanDime) [inline resources/js/Pages/Admin/Rooms/CreateRoomForm.vue:94]: The list of room types gets passed but it is hard coded here?

File: database/migrations/2021_01_20_031335_add_room_type_to_rooms_table.php (hunk 1 above)
Comment (joeyabou) [reply database/migrations/2021_01_20_031335_add_room_type_to_rooms_table.php:18]: i got an error when i tried to make it null so i changed it default and it worked fine.

File: resources/js/Pages/Admin/Rooms/CreateRoomForm.vue (hunk 2 above)
Comment (joeyabou) [reply resources/js/Pages/Admin/Rooms/CreateRoomForm.vue:94]: ur right it shouldnt be will address this

File: database/migrations/2021_01_20_031335_add_room_type_to_rooms_table.php (hunk 1 above)
Comment (EvanDime) [reply database/migrations/2021_01_20_031335_add_room_type_to_rooms_table.php:18]: if you want the possibility of this being null then add `->nullable()`.

Comment (EvanDime) [reply database/migrations/2021_01_20_031335_add_room_type_to_rooms_table.php:18]: right now the db will refuse to take null values unless nullable is specified

Comment (WindOfRussia) [reply database/migrations/2021_01_20_031335_add_room_type_to_rooms_table.php:18]: actually, the goal is that this is a NOT NULL field with no defaults. which is what failed when removing `->default('')`
postgres did not appreciate this extra validation

File: resources/js/Pages/Dashboard.vue [hunk 3]
```diff
@@ -2,18 +2,18 @@
...
-                    </div>
-                </div>
+                Dashboard
             </h2>
         </template>
 
+        <div class="py-12">
+            <div class="max-w-7xl mx-auto sm:px-6 lg:px-8">
+                <div class="bg-white overflow-hidden shadow-xl sm:rounded-lg">
+                    <a href="users">Users Page</a>
```
Comment (WindOfRussia) [inline resources/js/Pages/Dashboard.vue:12]: this href should probably be removed

**Review by WindOfRussia — APPROVED** [review]

File: resources/js/Pages/Dashboard.vue (hunk 3 above)
Comment (joeyabou) [reply resources/js/Pages/Dashboard.vue:12]: Should be fine since the larvel dashboard should be removed later on

**Review by EvanDime — APPROVED** [review]

**Pull request merged**